programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1998,12 +2038,87 @@ static void interrupt_handler(int sig) {
 }
 #endif
 
+size_t find_other_zets(model_list* ipcs, const char* ipc_base, const char* ipc_prefix) {
+    uv_fs_t fs;
+    int rc = uv_fs_scandir(uv_default_loop(), &fs, SOCKET_PATH, 0, NULL);
+    if (rc < 0) {
+        ZITI_LOG(ERROR, "failed to scan dir[%s]: %d/%s", ipc_base, rc, uv_strerror(rc));
+        return 0;
+    }
+    uv_dirent_t file;
+    while (uv_fs_scandir_next(&fs, &file) == 0) {
+        size_t len = strlen(ipc_prefix);
+        if(strncmp(file.name, ipc_prefix, len) == 0) {
+            model_list_append(ipcs, strdup(file.name));
+        }
+    }
+    return model_list_size(ipcs);
+}
+
+static bool same_dir(const char* path1, const char* path2) {
+    char resolved_path1[PATH_MAX];
+    char resolved_path2[PATH_MAX];
+
+    if (realpath(path1, resolved_path1) == NULL) {
+        ZITI_LOG(ERROR, "error resolving path1");
+        return 0;
+    }
+
+    if (realpath(path2, resolved_path2) == NULL) {
+        ZITI_LOG(ERROR, "error resolving path2");
+        return 0;
+    }
+
+    return strcmp(resolved_path1, resolved_path2) == 0;
+}
+
+static void configure_ipc(bool automatic_ipc_discriminator) {
+    if(ipc_discriminator == NULL && automatic_ipc_discriminator) {
+        int pid = getpid();
+        ipc_discriminator = calloc(10, sizeof(char));
+        snprintf(ipc_discriminator, 10, "%d", pid);
+    }
+
+    size_t socket_path_len = strlen(SOCKET_PATH);
+    size_t ipc_discriminator_len = 0;
+    if (ipc_discriminator != NULL) {
+        ipc_discriminator_len = strlen(ipc_discriminator);
+    }
+    size_t sockfilebase_len = strlen(sockfilebase);
+    size_t eventsockfilebase_len = strlen(eventsockfilebase);
+
+    sockfile = calloc(socket_path_len + sockfilebase_len + ipc_discriminator_len + 1, sizeof(char));
```

Comment:
why allocate just to print it below? however it is populated it can be logged directly

also this is a memory leak